pkg/ottl/contexts/internal/ottlcommon/metric.go

```diff
@@ -0,0 +1,223 @@
+	"AGGREGATION_TEMPORALITY_DELTA":          ottl.Enum(pmetric.AggregationTemporalityDelta),
+	"AGGREGATION_TEMPORALITY_CUMULATIVE":     ottl.Enum(pmetric.AggregationTemporalityCumulative),
+	"METRIC_DATA_TYPE_NONE":                  ottl.Enum(pmetric.MetricTypeEmpty),
+	"METRIC_DATA_TYPE_GAUGE":                 ottl.Enum(pmetric.MetricTypeGauge),
+	"METRIC_DATA_TYPE_SUM":                   ottl.Enum(pmetric.MetricTypeSum),
+	"METRIC_DATA_TYPE_HISTOGRAM":             ottl.Enum(pmetric.MetricTypeHistogram),
+	"METRIC_DATA_TYPE_EXPONENTIAL_HISTOGRAM": ottl.Enum(pmetric.MetricTypeExponentialHistogram),
+	"METRIC_DATA_TYPE_SUMMARY":               ottl.Enum(pmetric.MetricTypeSummary),
+}
+
+func MetricPathGetSetter[K MetricContext](path []ottl.Field) (ottl.GetSetter[K], error) {
+	if len(path) == 0 {
+		return accessMetric[K](), nil
+	}
+	switch path[0].Name {
+	case "name":
+		return accessName[K](), nil
+	case "description":
+		return accessDescription[K](), nil
+	case "unit":
+		return accessUnit[K](), nil
+	case "type":
+		return accessType[K](), nil
+	case "aggregation_temporality":
+		return accessAggTemporality[K](), nil
+	case "is_monotonic":
+		return accessIsMonotonic[K](), nil
+	case "data_points":
+		return accessDataPoints[K](), nil
+	}
+
+	return nil, fmt.Errorf("invalid metric path expression %v", path)
+}
+
+func accessMetric[K MetricContext]() ottl.StandardGetSetter[K] {
+	return ottl.StandardGetSetter[K]{
+		Getter: func(ctx K) interface{} {
+			return ctx.GetMetric()
+		},
+		Setter: func(ctx K, val interface{}) {
+			if newMetric, ok := val.(pmetric.Metric); ok {
+				newMetric.CopyTo(ctx.GetMetric())
+			}
+		},
+	}
+}
+
+func accessName[K MetricContext]() ottl.StandardGetSetter[K] {
+	return ottl.StandardGetSetter[K]{
+		Getter: func(ctx K) interface{} {
+			return ctx.GetMetric().Name()
+		},
+		Setter: func(ctx K, val interface{}) {
+			if str, ok := val.(string); ok {
+				ctx.GetMetric().SetName(str)
+			}
+		},
+	}
+}
+
+func accessDescription[K MetricContext]() ottl.StandardGetSetter[K] {
+	return ottl.StandardGetSetter[K]{
+		Getter: func(ctx K) interface{} {
+			return ctx.GetMetric().Description()
+		},
+		Setter: func(ctx K, val interface{}) {
+			if str, ok := val.(string); ok {
+				ctx.GetMetric().SetDescription(str)
+			}
+		},
+	}
+}
+
+func accessUnit[K MetricContext]() ottl.StandardGetSetter[K] {
+	return ottl.StandardGetSetter[K]{
+		Getter: func(ctx K) interface{} {
+			return ctx.GetMetric().Unit()
+		},
+		Setter: func(ctx K, val interface{}) {
+			if str, ok := val.(string); ok {
+				ctx.GetMetric().SetUnit(str)
+			}
+		},
+	}
+}
+
+func accessType[K MetricContext]() ottl.StandardGetSetter[K] {
+	return ottl.StandardGetSetter[K]{
+		Getter: func(ctx K) interface{} {
+			return int64(ctx.GetMetric().Type())
+		},
+		Setter: func(ctx K, val interface{}) {
+			// TODO Implement methods so correctly convert data types.
```

Comment:
It's not clear to me if this is intended for some future implementation or this is just not finished yet. It's not mentioned in the docs in a way I understand.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/15292